src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -152,16 +157,24 @@ public static unsafe string GetAsciiOrUTF8StringNonNullCharacters(this Span<byte
                         throw new InvalidOperationException();
                     }
 
-                    try
+                    if (useLatin1)
                     {
-                        resultString = HeaderValueEncoding.GetString(buffer, span.Length);
+                        StringUtilities.GetLatin1String(buffer, output, span.Length);
```

Comment:
Should take this opportunity to refactor this code not to create a `string` instance and mutate its contents (see line 145 above).

Reply:
We can get to that in 5.0. We'll want to limit changes for the 3.x patch.